modules/rest-util/test/blaze/middleware/fhir/output_test.clj

```diff
@@ -21,22 +23,42 @@
 
 (test/use-fixtures :each tu/fixture)
 
-(def resource-handler-200
+(def ^:private resource-handler-200
   "A handler which just returns a patient."
   (wrap-output
    (fn [_ respond _]
      (respond (ring/response {:fhir/type :fhir/Patient :id "0"})))))
 
-(def resource-handler-304
+(defn- binary-resource-handler-200
+  "A handler which uses the binary middleware and
+  returns a binary resource."
+  [{:keys [content-type data] :as _body}]
+  (wrap-binary-output
+   (fn [_ respond _]
+     (respond
+      (ring/response
+       (cond-> {:fhir/type :fhir/Binary}
+         data (assoc :data (type/base64Binary data))
+         content-type (assoc :contentType (type/code content-type))))))))
+
+(def ^:private resource-handler-304
   "A handler which returns a 304 Not Modified response."
   (wrap-output
    (fn [_ respond _]
      (respond (ring/status 304)))))
 
+(def ^:private binary-resource-handler-no-body
+  "A handler which uses the binary middleware and
+  returns a response with 200 and no body."
+  (wrap-binary-output
+   (fn [_ respond _]
+     (respond (ring/status 200)))))
+
 (defn- special-resource-handler [resource]
   (wrap-output
    (fn [_ respond _]
-     (respond (ring/response resource)))))
+     (respond (-> (ring/response resource)
+                  (ring/status 500))))))
```

Comment:
The Internal Server Error hat to come from the parsing error and not from the handler above the middleware chain.

Reply:
The code handling the XML generating error has to set the status to 500. In case the base64 encoding can throw an error, you should do the same.

Reply:
Then, I should add the corresponding regression tests, both for binary and non-binary resources, both for XML and JSON?

Reply:
Yes please add the tests for JSON, XML and Binary.